### DISCUSSION TITLE
Liveness and Readiness Fails with http-relative-path set

### DISCUSSION THREAD

**Discussion by gjain-sag** [keycloak/keycloak#12024] [Archive]
Hi Team,

As per our current Keycloak architecture we have a dependency on "/auth" relative path. I tried to retain the same via configuration "http-relative-path=/auth", once the cluster is up liveness and readiness fails with url " http://127.0.0.1:8080/health/live" configured.

Do we have any way to update the Liveness and Readiness configurations in the cluster based on http-relative-path??

**Comment by andreaTP** [comment]
Hi @gjain-sag , thanks for raising this discussion!

The `http-relative-path` field has been introduced to help in the migration from Wildfly to Quarkus distribution, since the new Operator does support only the latter I would be inclined to actually "remove" this field from the `serverConfiguration`.

You will be still able to set it in a few ways, but I do believe that we should avoid supporting compatibility layers whenever possible.

Maybe I'm missing something, would you like to share the reasons why it's hard for you to remove the relative path?

cc. @pedroigor @vmuzikar

**Reply by vmuzikar** [reply]
@gjain-sag Thank you for raising this topic! Even though we don't support direct migration from WF dist (and the old Operator) to Quarkus dist (and the new Operator), I believe it is still a valid use case to have the possibility to use `/auth` relative path. Otherwise we would force the users to actually update their apps to use a new address which would introduce additional migration costs and is not strictly necessary.

Created https://github.com/keycloak/keycloak/issues/12100.

**Reply by andreaTP** [reply]
Before tackling this Issue I would like to hear more opinions on this one, I believe that @pedroigor and I are more inclined to drop the support instead.

**Reply by gjain-sag** [reply]
Thanks @andreaTP ,

We have many dependent applications which are using our client for authentication and other configurations. Removing support for "/auth" will create backward compatibility issues for us.

**Reply by psytester** [reply]
In our case we have one reverse proxy TLS endpoint for one public sub domain, serving all calls to different backends based of url path. Removing `/auth` would require an additional public sub domain just for the keycloak part.

At the end it's just more PKI and system maintainance for nothing.

**Comment by omasseau** [comment]
Same for us. Removing the /auth path would require us to adapt all our reverse-proxies configurations and update some of our apps.
This could be very touchy.

**Comment by pedroigor** [comment]
Looks like it should be there for a while but we should have a deadline for removing it. @stianst wdyt?
